src/Raven.Server/Documents/Queries/Graph/SingleEdgeMatcher.cs

```diff
@@ -81,6 +78,93 @@ private void ThrowMissingEdgeProjection()
         }
 
 
+        private void EdgeMatch(Match match, BlittableJsonReaderObject blittableJsonReaderObject)
+        {
+            //projection
+            if (Edge.Where == null)
+            {
+                AddEdgeAfterFiltering(match, blittableJsonReaderObject, Edge.Project.FieldValue);
+                return;
+            }
+            if (Edge.Where is MethodExpression method)
+            {
+                if (EdgeMatchExact(blittableJsonReaderObject, method))
+                {
+                    AddEdgeAfterFiltering(match, blittableJsonReaderObject, Edge.Project?.FieldValue);
+                }
+                return;
+            }
+            if (Edge.Where.IsMatchedBy(blittableJsonReaderObject, QueryParameters) != false)
+            {
+                AddEdgeAfterFiltering(match, blittableJsonReaderObject, Edge.Project?.FieldValue);
+            }
+        }
+
+        private bool EdgeMatchExact(BlittableJsonReaderObject blittableJsonReaderObject, MethodExpression methodExpression)
+        {
+            if (string.Equals(methodExpression.Name.Value, "exact", StringComparison.OrdinalIgnoreCase) == false)
+                throw new NotSupportedException($"Where clause on edge doesn't support {methodExpression} method expression");
+            var arg = methodExpression.Arguments[0] as BinaryExpression;
+            var left = arg.Left as FieldExpression;
+            var right = arg.Right as ValueExpression;
+            if (arg == null || (arg.Operator != OperatorType.Equal && arg.Operator != OperatorType.NotEqual) || left == null || right == null)
```

Comment:
That isn't the right way to handle this.
We need to pass options to `IsMatchBy` where we'll configure `Exact=true` and change the behavior here: https://github.com/ravendb/ravendb/blob/v4.2/src/Raven.Server/Documents/Queries/AST/QueryExpression.cs#L129

Reply:
This doesn't handle methods methods are not suppose to be handled at that code path i don't understand why this come up now